pycargoebuild/cargo.py

```diff
@@ -117,7 +135,11 @@ def get_package_directory(self, distdir: Path) -> PurePath:
     def get_git_crate_entry(self, distdir: Path) -> str:
         subdir = (str(self.get_package_directory(distdir))
                   .replace(self.commit, "%commit%"))
-        return f"{self.repository};{self.commit};{subdir}"
+        if self.repo_ext:
```

Comment:
Please don't make conditionals on side effects.

Reply:
This is still checking for suffix rather than matching against `self.repo_host`.